packages/@divvi/mobile/src/identity/smsRetrieval.ts

```diff
@@ -12,41 +12,41 @@ export interface SmsEvent {
 export async function startSmsRetriever() {
   Logger.debug(TAG + '@SmsRetriever', 'Starting sms retriever')
   try {
-    const result = await SmsRetriever.startSmsRetriever()
-    if (result) {
+    // startOtpListener starts the SMS listener and returns a subscription
+    const subscription = await startOtpListener((message: string) => {
+      Logger.debug(TAG + '@SmsRetriever', 'Message Received from sms listener', message)
+    })
+    if (subscription) {
       Logger.debug(TAG + '@SmsRetriever', 'Retriever started successfully')
+      return true
     } else {
       Logger.error(TAG + '@SmsRetriever', 'Start retriever reported failure')
+      return false
     }
   } catch (error) {
     Logger.error(TAG + '@SmsRetriever', 'Error starting retriever', error)
+    return false
   }
 }
 
 export function addSmsListener(onSmsRetrieved: (message: SmsEvent) => void) {
   Logger.debug(TAG + '@SmsRetriever', 'Adding sms listener')
   try {
-    void SmsRetriever.addSmsListener((event: SmsEvent) => {
-      if (!event) {
+    // addListener adds a callback to receive SMS messages
+    const subscription = addListener((message: string) => {
+      if (!message) {
         Logger.error(TAG + '@SmsRetriever', 'Sms listener event is null')
         return
       }
-      if (event.error) {
-        Logger.error(TAG + '@SmsRetriever', 'Sms listener error: ' + event.error)
-        return
-      }
-      if (event.timeout) {
-        Logger.warn(TAG + '@SmsRetriever', 'Sms listener timed out')
-        return
-      }
-      if (!event.message) {
-        Logger.warn(TAG + '@SmsRetriever', 'Sms listener returned empty message')
-        return
-      }
 
-      Logger.debug(TAG + '@SmsRetriever', 'Message Received from sms listener', event.message)
+      Logger.debug(TAG + '@SmsRetriever', 'Message Received from sms listener', message)
+      // Create a compatible event object
+      const event: SmsEvent = { message }
       onSmsRetrieved(event)
     })
+
+    // Store subscription for cleanup if needed
```

Comment:
Does this code end up using this?  Dumb question: where is it stored (and how is it removed?)?

Reply:
We're fine to remove this. The cleanup appears to be handled by `removeListener()`